qutebrowser/browser/network/networkmanager.py

```diff
@@ -358,6 +358,23 @@ def createRequest(self, op, req, outgoing_data):
             dnt = '0'.encode('ascii')
         req.setRawHeader('DNT'.encode('ascii'), dnt)
         req.setRawHeader('X-Do-Not-Track'.encode('ascii'), dnt)
+
+        def same_domain():
+            tabbed_browser = objreg.get('tabbed-browser', scope='window', window=self._win_id)
+            view = tabbed_browser.currentWidget()
+            print(req.url().host() == view.url().host(), repr(view.url().host()), repr(req.url().host()))
+            # TODO: We probably want to allow headers if we're on test.com and
+            # doing a request to www.test.com? Or maybe a new settings value for
+            # this?
+            return req.url().host() == view.url().host()
```

Comment:
I guess we have to find out how to do this comparison, yeah.

A far more restrictive possibility would be using `urlutils.host_tuple()`, i.e. scheme/host/port, but I think this is definitely too restrictive here.

I wonder if there's a way to get the base domain based on a subdomain with Python/Qt built in, otherwise it might make sense to bundle the [Mozilla Public Suffix List](https://publicsuffix.org/list/) for that.